src/editor/ui_menus/scenario_tool_infrastructure_options_menu.cc

```diff
@@ -0,0 +1,407 @@
+					                    "conventions, identified %3$s as headquarters")) %
+					    p % player.tribe().descname().c_str() %
+					    egbase.tribes().get_building_descr(di)->name().c_str())
+					      .str());
+				}
+			}
+		}
+		if (index != Widelands::INVALID_INDEX) {
+			const Widelands::BuildingDescr& b = *egbase.tribes().get_building_descr(index);
+			if (b.suitability(
+			       egbase.map(), egbase.map().get_fcoords(egbase.map().get_starting_pos(p)))) {
+				player.force_building(egbase.map().get_starting_pos(p), {std::make_pair(index, "")});
+			} else {
+				errors.push_back(
+				   (boost::format(
+				       _("Player %1$u (tribe %2$s): Starting position is not suited for a %3$s")) %
+				    p % player.tribe().descname().c_str() % b.name().c_str())
+				      .str());
+			}
+		} else {
+			errors.push_back(
+			   (boost::format(
+			       _("Player %1$u (tribe %2$s): Headquarters building could not be identified")) %
+			    p % player.tribe().descname().c_str())
+			      .str());
+		}
+	}
+	std::string text;
+	if (errors.empty()) {
+		text =
+		   _("All buildings placed successfully. Don’t forget to add the initial wares and workers.");
+	} else {
+		text = _("Not all buildings could be placed!");
+		text += "\n";
+		text += (boost::format(
+		            ngettext("%u warning generated:", "%u warnings generated:", errors.size())) %
+		         errors.size())
+		           .str();
+		text += "\n";
+		for (const std::string& e : errors) {
+			text += "\n" + e;
+		}
+	}
+	UI::WLMessageBox m(get_parent(), errors.empty() ? _("Success") : _("Error"), text,
+	                   UI::WLMessageBox::MBoxType::kOk);
+	m.run<UI::Panel::Returncodes>();
+}
+
+void ScenarioToolInfrastructureOptionsMenu::select_tab() {
+	Widelands::EditorGameBase& egbase = eia().egbase();
+
+	const Widelands::PlayerNumber p = tabs_.active() + 1;
+	assert(p);
+	assert(p <= egbase.map().get_nrplayers());
+	tool_.set_player(p);
+
+	const Widelands::TribeDescr& tribe = *egbase.tribes().get_tribe_descr(
+	   egbase.tribes().tribe_index(egbase.map().get_scenario_player_tribe(p)));
+	auto& list = tool_.get_index();
+	for (auto it = list.begin(); it != list.end();) {
+		bool erase = false;
+		switch (it->first) {
+		case Widelands::MapObjectType::FLAG:
+			break;
+		case Widelands::MapObjectType::IMMOVABLE:
+			erase = !tribe.has_immovable(it->second);
+			break;
+		case Widelands::MapObjectType::BUILDING:
+			erase =
+			   !tribe.has_building(it->second) &&
+			   tribe.get_building_descr(it->second)->type() != Widelands::MapObjectType::MILITARYSITE;
+			break;
+		default:
+			NEVER_HERE();
+		}
+		if (erase)
+			it = list.erase(it);
+		else
+			++it;
+	}
+
+	update_text();
+	select_correct_tool();
+}
+
+void ScenarioToolInfrastructureOptionsMenu::update_text() {
+	const size_t nr_items = tool_.get_index().size();
+	if (nr_items == 0) {
+		selected_items_.set_text(_("Nothing selected"));
```

Comment:
We should unify this text with `pgettext("dropdown", "Not Selected");` in the dropdowns to have less work for translators.